cg/meta/workflow/nf_analysis.py

```diff
@@ -307,3 +320,109 @@ def get_deliverables_for_case(self, case_id: str) -> PipelineDeliverables:
                 )
             files.append(FileDeliverable(**file))
         return PipelineDeliverables(files=files)
+
+    def get_multiqc_json_path(self, case_id: str) -> Path:
+        """Return the path of the multiqc_data.json file."""
+        return Path(
+            self.root_dir,
+            case_id,
+            MultiQC.MULTIQC,
+            MultiQC.MULTIQC_DATA,
+            MultiQC.MULTIQC_DATA + FileExtensions.JSON,
+        )
+
+    def get_pipeline_metrics(self) -> dict:
+        """Get nf-core pipeline metrics constants."""
+        return {}
+
+    def get_multiqc_json_metrics(self, case_id: str) -> list[MetricsBase]:
+        """Get a multiqc_data.json file and returns metrics and values formatted."""
+        case: Case = self.status_db.get_case_by_internal_id(internal_id=case_id)
+        sample_id: str = case.links[0].sample.internal_id
+        multiqc_json = MultiqcDataJson(
+            **read_json(file_path=self.get_multiqc_json_path(case_id=case_id))
+        )
+        metrics_values: dict = {}
+
+        for key in multiqc_json.report_general_stats_data:
+            if case_id in key:
+                metrics_values.update(list(key.values())[0])
+                LOG.info(f"Key: {key}, Values: {list(key.values())[0]}")
+
+        return [
+            MetricsBase(
+                header=None,
+                id=sample_id,
+                input=MultiQC.MULTIQC_DATA + FileExtensions.JSON,
+                name=metric_name,
+                step=MultiQC.MULTIQC,
+                value=metric_value,
+                condition=self.get_pipeline_metrics().get(metric_name, None),
+            )
+            for metric_name, metric_value in metrics_values.items()
+        ]
+
+    def get_multiqc_json_metrics_per_sample(self, case_id: str) -> list[MetricsBase]:
+        """Return a list of the metrics specified in a MultiQC json file for the case samples."""
+        multiqc_json: list[dict] = MultiqcDataJson(
+            **read_json(file_path=self.get_multiqc_json_path(case_id=case_id))
+        ).report_general_stats_data
+        samples: list[Samples] = self.status_db.get_samples_by_case_id(case_id=case_id)
```

Comment:
maybe this was my bad, sorry
```suggestion
        samples: list[Sample] = self.status_db.get_samples_by_case_id(case_id=case_id)
```